cli/src/lib.rs

```diff
@@ -107,36 +102,41 @@ impl Drop for Iroha {
     }
 }
 
-struct FromNetworkBaseRelay {
+struct NetworkRelay {
     sumeragi: Arc<Sumeragi>,
-    broker: Broker,
+    block_sync: BlockSynchronizerHandle,
+    network: IrohaNetwork,
+    shutdown_notify: Arc<Notify>,
     #[cfg(debug_assertions)]
     freeze_status: Arc<AtomicBool>,
 }
 
-#[async_trait::async_trait]
-impl Actor for FromNetworkBaseRelay {
-    async fn on_start(&mut self, ctx: &mut iroha_actor::prelude::Context<Self>) {
-        // to start connections
-        self.broker.subscribe::<NetworkBaseRelayOnlinePeers, _>(ctx);
-        self.broker.subscribe::<iroha_core::NetworkMessage, _>(ctx);
+impl NetworkRelay {
+    fn start(self) {
+        tokio::task::spawn(self.run());
     }
-}
-
-#[async_trait::async_trait]
-impl Handler<NetworkBaseRelayOnlinePeers> for FromNetworkBaseRelay {
-    type Result = ();
 
-    async fn handle(&mut self, msg: NetworkBaseRelayOnlinePeers) {
-        self.sumeragi.update_online_peers(msg.online_peers);
+    async fn run(mut self) {
+        let (sender, mut receiver) = mpsc::channel(1);
+        self.network.subscribe_t(sender).await;
+        #[allow(clippy::redundant_pub_crate)]
```

Comment:
Remember to create an issue in `tokio` to fix this.

Reply:
Opened tokio-rs/tokio#5616

Reply:
Update, opened rust-lang/rust-clippy#10636 on clippy side.